testing/e2e/manifests/common.go

```diff
@@ -61,6 +64,30 @@ func setGroupKindVersion(obj client.Object) {
 
 // newGoDeployment creates a new basic Go deployment with a single main.go file from contents
 func newGoDeployment(contents, namespace, name string) *appsv1.Deployment {
+	command := []string{
+		"/bin/sh",
+		"-c",
+		"mkdir source && cd source && go mod init source && echo '" + contents + "' > main.go && go run main.go",
+	}
+
+	// check if contents contains non standard library imports.
+	// if it does, we need a different command
+	fset := token.NewFileSet()
+	f, err := parser.ParseFile(fset, "", contents, parser.ImportsOnly)
+	if err != nil {
+		panic(fmt.Errorf("parsing contents: %w", err)) // this should only happen if the contents are invalid Go code
+	}
+	for _, imp := range f.Imports {
+		if strings.Contains(imp.Path.Value, ".") { // if it contains a dot it's not a standard library import
+			command = []string{
+				"/bin/sh",
+				"-c",
+				"mkdir source && cd source && go mod init source && echo '" + contents + "' > main.go && go mod tidy && go run main.go",
```

Comment:
any reason we dont just use this the command definition above?

Reply:
`go mod tidy` installs dependencies as well. We check for if the thing we are trying to run has any external dependencies that need to be installed and if so we add that "install" command to the script. Go mod tidy just happens to be how we are choosing to install in this case.

Reply:
Exactly, so you can use a single command definition that includes `go mod tidy` and remove the package checks.  There wouldnt be any real performance hits if something doesnt have external dependencies